packages/autocomplete-plugin-algolia-insights/src/createAlgoliaInsightsPlugin.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  AutocompletePlugin,
+  AutocompleteState,
+} from '@algolia/autocomplete-core';
+import { createRef, debounce, isEqual } from '@algolia/autocomplete-shared';
+
+import { createClickedEvent } from './createClickedEvent';
+import { createSearchInsightsApi } from './createSearchInsightsApi';
+import { createViewedEvents } from './createViewedEvents';
+import { isAlgoliaInsightsHit } from './isAlgoliaInsightsHit';
+import {
+  AlgoliaInsightsHit,
+  InsightsApi,
+  InsightsClient,
+  OnHighlightParams,
+  OnItemsChangeParams,
+  OnSelectParams,
+} from './types';
+
+const VIEW_EVENT_DELAY = 400;
```

Comment:
I wonder if this should be a parameter to createAlgoliaInsightsPlugin

Reply:
I think we should wait for users feedback first. Ideally we should find the right value that fits everybody.